e2e-tests/support/tasks.js

```diff
@@ -81,14 +87,22 @@ module.exports = {
         );
       },
 
+      /**
+       * Inserts utilisation report details to the SQL database
+       * @param {Partial<import('@ukef/dtfs2-common').UtilisationReportEntity[]>} utilisationReportDetails
+       * @returns {import('@ukef/dtfs2-common').UtilisationReportEntity[]} The inserted reports
+       */
       async insertUtilisationReportDetailsIntoDb(utilisationReportDetails) {
-        const utilisationReports = await db.getCollection(DB_COLLECTIONS.UTILISATION_REPORTS);
-        return utilisationReports.insertMany(utilisationReportDetails);
+        const utilisationReportRepo = SqlDbDataSource.getRepository(UtilisationReportEntity);
+        const saveOperations = utilisationReportDetails.map((utilisationReport) => utilisationReportRepo.save(utilisationReport));
+        return await Promise.all(saveOperations);
```

Comment:
Is this again because of the different types of reports we can save at once?

Reply:
Yup